crates/matrix-sdk/tests/integration/room/attachment/mod.rs

```diff
@@ -290,3 +292,230 @@ async fn test_room_attachment_send_mentions() {
 
     assert_eq!(event_id!("$h29iv0s8:example.com"), response.event_id)
 }
+
+#[cfg(feature = "image-proc")]
+const IMAGE_BYTES: &[u8] = include_bytes!("matrix-rusty.jpg");
+
+#[cfg(feature = "image-proc")]
+#[async_test]
+async fn test_room_attachment_generate_thumbnail_original_format() {
+    let (client, server) = logged_in_client_with_server().await;
+
+    Mock::given(method("PUT"))
+        .and(path_regex(r"^/_matrix/client/r0/rooms/.*/send/.*"))
+        .and(header("authorization", "Bearer 1234"))
+        .and(body_partial_json(json!({
+            "url": "mxc://localhost/AQwafuaFswefuhsfAFAgsw",
+            "info": {
+                "mimetype": "image/jpeg",
+                "thumbnail_info": {
+                    "h": 600,
+                    "w": 600,
+                    "mimetype":"image/jpeg",
+                },
+                "thumbnail_url": "mxc://localhost/AQwafuaFswefuhsfAFAgsw",
+            }
+        })))
+        .respond_with(ResponseTemplate::new(200).set_body_json(&*test_json::EVENT_ID))
+        .mount(&server)
+        .await;
+
+    Mock::given(method("POST"))
+        .and(path("/_matrix/media/r0/upload"))
+        .and(header("authorization", "Bearer 1234"))
+        .and(header("content-type", "image/jpeg"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(json!({
+          "content_uri": "mxc://localhost/AQwafuaFswefuhsfAFAgsw"
+        })))
+        .expect(2)
+        .mount(&server)
+        .await;
+
+    mock_sync(&server, &*test_json::SYNC, None).await;
+    mock_encryption_state(&server, false).await;
+
+    let sync_settings = SyncSettings::new().timeout(Duration::from_millis(3000));
+
+    let _response = client.sync_once(sync_settings).await.unwrap();
+
+    let room = client.get_room(&DEFAULT_TEST_ROOM_ID).unwrap();
+
+    let config = AttachmentConfig::new().generate_thumbnail(None, ThumbnailFormat::Original);
+
+    let response = room
+        .send_attachment("image", &mime::IMAGE_JPEG, IMAGE_BYTES.to_vec(), config)
+        .await
+        .unwrap();
+
+    assert_eq!(event_id!("$h29iv0s8:example.com"), response.event_id)
```

Comment:
Here and in the 3 other tests: can we replace the raw `event_id` with `test_json::EVENT_ID` too?

Reply:
`test_json::EVENT_ID` is a `serde_json::Value` containing a JSON object, so we can't use it for asserting the event ID, afaiu.

We would probably need a `matrix_sdk_test::DEFAULT_TEST_EVENT_ID`, like we have for the room ID for that.